### DISCUSSION TITLE
Override Named Anchor Spin Behavior

### DISCUSSION THREAD

**Discussion by AUnicyclingProgrammer** [BelfrySCAD/BOSL2#1400] [Q&A]
I have a custom part that I've made with a mounting hinge that I want to attach to a larger assembly. I've added a named anchor to the exact center-point of the mounting hinge, as I need to be able attach this model to another object using that anchor.

The problem I'm running into is as follows:
I like the current orientation of all the model anchors except for the named anchor, and would like to preserve the default spin behavior for the default anchors. I want the hinge anchor to act differently from the other anchors. If the model is using that anchor, it want it to be rotate the around the hinge as it would in the real world.

Is there a way to over-ride the spin behavior only if you are using a particular anchor using the attachables system? If so, how do you accomplish that?

Thank you.

Part in question:
![image](https://github.com/BelfrySCAD/BOSL2/assets/66497241/7a1963c6-a9e8-42fc-88db-1d0a0879339a)

**Comment by adrianVmariano** [comment]
Sounds like you want the spin parameter for the "hinge" anchor to spin on a different axis than the normal one.  There is no way to change the spin axis.  I think the way to do what you want is to age a hingespin parameter to your model that executes the desired rotation.

**Comment by AUnicyclingProgrammer** [comment]
That is correct, I wanted the spin parameter to rotate on a different axis than the normal one while still preserving the model orientation. I solved this problem in a similar way to what you suggested. I knew I would never use the normal spin behavior, so I manually intercepted and modified the model orientation if it was using the "hinge" anchor. It's a duct-tape fix, but that's all I needed to move on with my project.

I'm still debating if this kind of functionality would ever be needed enough to be worth implementing. I've never run into this problem since then, but I also haven't worked with hinges since. This type of behavior doesn't seem like it would typically be worth implementing, except if you were working with hinges (or another object that rotates around a point in a similar way), in which case it would be a very useful feature.

**Reply by adrianVmariano** [reply]
I think that changing the spin parameter to sometimes do something different would be fairly complicated with align(), which looks at the spin to adjust the anchors---though since it doesn't arise for named anchors that wouldn't be an issue.  Generally though I'd think that adding a new parameter is probably a better answer than reusing spin sometimes, which could cause trouble if for some reason someone tries to use spin the normal way---though perhaps not an issue if the module is just for your personal use.  For obvious reasons I am generally thinking when I construct modules about general use...and you never know what users might find a use for.  Revar added a special gearspin parameter for gears to make them rotate around their centers even when they are anchored elsewhere, so there is a precedent for this sort of thing.
